CRM/Contribute/Page/ContributionRecurTab.php

```diff
@@ -0,0 +1,493 @@
+        ),
+        CRM_Core_Action::UPDATE => array(
+          'name' => ts('Edit'),
+          'title' => ts('Edit Recurring Payment'),
+          'url' => 'civicrm/contributionrecur/add',
+          'qs' => "reset=1&action=update&crid=%%crid%%&cid=%%cid%%&context={$context}",
+        ),
+        CRM_Core_Action::DELETE => array(
+          'name' => ts('Delete'),
+          'title' => ts('Delete Recurring Payment'),
+          'url' => 'civicrm/contributionrecur/add',
+          'qs' => "reset=1&action=delete&crid=%%crid%%&cid=%%cid%%&context={$context}",
+        ),
+        CRM_Core_Action::DISABLE => array(
+          'name' => ts('Cancel'),
+          'title' => ts('Cancel'),
+          'ref' => 'crm-enable-disable',
+        ),
+      );
+    }
+    
+    if ($recurID) {
+      $paymentProcessorObj = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($recurID, 'recur', 'obj');
+      if (is_object($paymentProcessorObj) && $paymentProcessorObj->isSupported('cancelSubscription')) {
+        unset(self::$_links[CRM_Core_Action::DISABLE]['extra'], self::$_links[CRM_Core_Action::DISABLE]['ref']);
+        self::$_links[CRM_Core_Action::DISABLE]['url'] = "civicrm/contribute/unsubscribe";
+        self::$_links[CRM_Core_Action::DISABLE]['qs'] = "reset=1&crid=%%crid%%&cid=%%cid%%&context={$context}";
+      }
+      if (is_object($paymentProcessorObj) && $paymentProcessorObj->isSupported('updateSubscriptionBillingInfo')) {
+        self::$_links[CRM_Core_Action::RENEW] = array(
+          'name' => ts('Change Billing Details'),
+          'title' => ts('Change Billing Details'),
+          'url' => 'civicrm/contribute/updatebilling',
+          'qs' => "reset=1&crid=%%crid%%&cid=%%cid%%&context={$context}",
+        );
+      }
+    }
+
+    return self::$_links;
+  }
+  // end function
+
+  /**
+   * called when action is browse.
+   *
+   */
+  public function browse() {
+    // add annual contribution
+    $annual = array();
+    list($annual['count'],
+      $annual['amount'],
+      $annual['avg']
+      ) = CRM_Contribute_BAO_Contribution::annual($this->_contactId);
+    $this->assign('annual', $annual);
+
+    $controller = new CRM_Core_Controller_Simple(
+      'CRM_Contribute_Form_Search',
+      ts('Contributions'),
+      $this->_action,
+      FALSE, FALSE, TRUE
+    );
+    $controller->setEmbedded(TRUE);
+    $controller->reset();
+    $controller->set('cid', $this->_contactId);
+    $controller->set('crid', $this->_crid);
+    $controller->set('context', 'contribution');
+    $controller->set('limit', 50);
+    $controller->process();
+    $controller->run();
+
+    // add recurring block
+    $action = array_sum(array_keys($this->recurLinks()));
+    $params = CRM_Contribute_BAO_ContributionRecur::getRecurContributions($this->_contactId);
+
+    // Get all backoffice payment processors
+    $backOfficePaymentProcessors = CRM_Contribute_BAO_ContributionRecur::getBackOfficePaymentProcessors();
```

Comment:
Since this doesn't extend form the previous comments don't apply but we can
 still use CRM_Financial_BAO_PaymentProcessor::getPaymentProcessors(array('BackOffice'));

I would note it might be worth extending the function below to support checking for recur capability ie. 

$hasProc = CRM_Financial_BAO_PaymentProcessor::hasPaymentProcessorSupporting(array('BackOffice'));
$hasProc = CRM_Financial_BAO_PaymentProcessor::hasPaymentProcessorSupporting(array('BackOffice', 'Recur'));